cocotb/triggers.py

```diff
@@ -656,12 +641,11 @@ def __init__(self, *triggers):
         self.triggers = tuple(triggers)
 
         # Do some basic type-checking up front, rather than waiting until we
-        # yield them.
-        allowed_types = (Trigger, Waitable, decorators.RunningTask)
+        # await them.
         for trigger in self.triggers:
-            if not isinstance(trigger, allowed_types):
+            if not inspect.isawaitable(trigger):
```

Comment:
This simplification allows the use of `types.coroutine` and other awaitable objects. We don't need to be so strict about types anymore because we bypass the scheduler.

Reply:
This changes the behavior of `First` and `Combine`, this was reverted.